### PR TITLE
ENHANCE: Performance improvements to NormalizerViaRadical

### PR DESCRIPTION
**Pull request by hulpke**
- Avoid unnecessary membership test that would be expensive
- Clean up commented-out debugging code
- Improve action on cohomology by going to factor modules first

### CODE REVIEW

File: lib/norad.gi [hunk 1]
```diff
@@ -937,174 +878,169 @@ local sus,ser,len,factorhom,uf,n,d,up,mran,nran,mpcgs,pcgs,pcisom,nf,ng,np,sub,
...
+        sub:=List(sumos[p1],x->PcElementByExponents(famo,x));
+        sub:=InducedPcgsByGeneratorsNC(NumeratorOfModuloPcgs(famo),Concatenation(DenominatorOfModuloPcgs(famo),sub));
+        lfamo:=NumeratorOfModuloPcgs(famo) mod sub;
+
+        Info(InfoFitFree,3,"Factor ",p1," Module ",Length(lfamo));
+
+        ranges:=List([1..Length(part)],
+          x->[(x-1)*Length(lfamo)+1..x*Length(lfamo)]);
+
+        sub:=Concatenation(List(part,a->List(lfamo,x->Zero(f))));
```
[the remark this answers is not in the document]
Comment (hulpke) [reply lib/norad.gi]: No. It is `ListWithIdenticalEntries(Length(part)*Length(lfamo),Zero(f))`. The `Concatenation` makes it clear that this is cohomology liosted as module entries associated to the generators in `part`.

File: lib/norad.gi [hunk 2]
```diff
@@ -937,174 +878,169 @@ local sus,ser,len,factorhom,uf,n,d,up,mran,nran,mpcgs,pcgs,pcisom,nf,ng,np,sub,
...
+          #l:=List([1..Length(part)],x->LeftQuotient(part[x],l[x]));
+          #l:=List(l,x->ExponentsOfPcElement(famo,x)*One(f));
+
+          l:=List([1..Length(part)],
+            x->ExponentsOfPcElement(lfamo, LeftQuotient(part[x],l[x]))*One(f));
+
+          l:=Concatenation(l);
+          l:=SiftedVector(bound,l);
+          ConvertToVectorRep(l,Size(f));
+          MakeImmutable(l);
```
[the remark this answers is not in the document]
Comment (hulpke) [reply lib/norad.gi]: Done. (Code predated `ImmutableVector` and the change here was indent or similar.)

File: lib/norad.gi [hunk 3]
```diff
@@ -937,174 +878,169 @@ local sus,ser,len,factorhom,uf,n,d,up,mran,nran,mpcgs,pcgs,pcisom,nf,ng,np,sub,
...
+        # coboundaries
+        boundbas:=List(lfamo,x->Concatenation(List(part,
+                  y->ExponentsOfPcElement(lfamo,Comm(y,x))*One(f))));
+        boundbas:=ImmutableMatrix(f,boundbas);
+        bound:=List(boundbas,ShallowCopy);
+        TriangulizeMat(bound);
+        bound:=List(bound,Zero);
+        bound:=Basis(VectorSpace(f,bound));
+
+  #if
```
[the remark this answers is not in the document]
Comment (hulpke) [reply lib/norad.gi]: There should at some point be a case distinction.

File: lib/norad.gi (hunk 1 above)
[the remark this answers is not in the document]
Comment (fingolfin) [reply lib/norad.gi]: Indeed, somehow I (rather stupidly) missed the `Concatenation` bit.

**Pull request merged**